sympy/unify/core.py

```diff
@@ -1,179 +1,358 @@
-""" Generic Unification algorithm for expression trees with lists of children
+"""Unification algorithm
 
-This implementation is a direct translation of
+The original version (but with some correctness issues) can be found here
+http://aima.cs.berkeley.edu/python/logic.html
 
-Artificial Intelligence: A Modern Approach by Stuart Russel and Peter Norvig
-Second edition, section 9.2, page 276
+We extended the original algorithm with the capability to
 
-It is modified in the following ways:
+- Handle associative-commutative unification capability.
+- Designed type system to make this work across almosst every objects of
+  python by composition.
```

Comment:
```suggestion
- Designed type system to make this work with almost all Python objects by composition.
```

Reply:
(not exactly sure what you mean so this might still not quite capture the intent)